framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,417 @@
+     *
+     * @param right the range of bits by which this range is left shifted
+     * @return the range resulting from left shifting this range by the specified range
+     */
+    public Range shiftLeft(Range right) {
+        long resultFrom = from << (from >= 0 ? right.from : right.to);
+        long resultTo = to << (to >= 0 ? right.to : right.from);
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Returns the smallest range that includes all possible values resulting from signed right
+     * shifting an arbitrary value in this range by an arbitrary number of bits in the specified
+     * range. We call this the signed right shift operation of a range.
+     *
+     * @param right the range of bits by which this range is signed right shifted
+     * @return the range resulting from signed right shifting this range by the specified range
+     */
+    public Range signedShiftRight(Range right) {
+        long resultFrom = from >> (from >= 0 ? right.to : right.from);
+        long resultTo = to >> (to >= 0 ? right.from : right.to);
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Returns the range of a variable that falls within this range after applying the unary plus
+     * operation (which is a no-op).
+     *
+     * @return this range
+     */
+    public Range unaryPlus() {
+        return this;
+    }
+
+    /**
+     * Returns the range of a variable that falls within this range after applying unary minus
+     * operation.
+     *
+     * @return the resulted range of applying unary minus on an arbitrary value in this range
+     */
+    public Range unaryMinus() {
+        return new Range(-to, -from);
+    }
+
+    /**
+     * Returns the range of a variable that falls within this range after applying bitwise
+     * complement operation.
+     *
+     * @return the resulted range of applying bitwise complement on an arbitrary value in this range
+     */
+    public Range bitwiseComplement() {
+        return new Range(~to, ~from);
+    }
+
+    /**
+     * Determines a refined range of a variable that fall within this {@code Range} under the
+     * condition that this variable is less than another variable that fall within the specified
+     * {@code Range}. This is used for calculating the control flow refined result of the &lt;
+     * operator. i.e.
+     *
+     * <pre>
+     * <code>
+     *     {@literal @}IntRange(from = 0, to = 10) int a;
+     *     {@literal @}IntRange(from = 3, to = 5) int b;
+     *     ...
+     *     if (a &lt; b) {
+     *         // range of <i>a</i> is now refined to [0, 4] because a value in range [5, 10]
+     *         // cannot be smaller than variable <i>b</i> with range [3, 5]
+     *         ...
+     *     }
+     * </code>
+     * </pre>
+     *
+     * @param right the specified {@code Range} to compare with.
+     * @return the refined {@code Range}.
+     */
+    public Range lessThan(Range right) {
```

Comment:
The name "lessThan" is a bit misleading.
When I see the code
```range1.lessThan(range2)```
my first thought is that the operator is a conditional.
It's actually doing refinement, so I suggest that this and related classes be renamed.
Some possibilities:
 * `refineLessThan`
 * `andIsLessThan`

Maybe you have better ideas for a new name.